src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveRenderedService.cs

```diff
@@ -0,0 +1,140 @@
+    {
+        EnsureArg.IsNotNull(request, nameof(request));
+
+        // To keep track of how long render operation is taking
+        Stopwatch sw = new Stopwatch();
+
+        int partitionKey = _dicomRequestContextAccessor.RequestContext.GetPartitionKey();
+        AcceptHeader returnHeader = GetValidRenderAcceptHeader(request.AcceptHeaders);
+
+        try
+        {
+            // this call throws NotFound when zero instance found
+            InstanceMetadata instance = (await _instanceStore.GetInstancesWithProperties(
+                ResourceType.Instance, partitionKey, request.StudyInstanceUid, request.SeriesInstanceUid, request.SopInstanceUid, cancellationToken)).First();
+
+            FileProperties fileProperties = await RetrieveHelpers.CheckFileSize(_blobDataStore, _retrieveConfiguration.MaxDicomFileSize, instance.VersionedInstanceIdentifier.Version, true, cancellationToken);
+            Stream stream = await _blobDataStore.GetFileAsync(instance.VersionedInstanceIdentifier.Version, cancellationToken);
+            sw.Start();
+
+            DicomFile dicomFile = await DicomFile.OpenAsync(stream, FileReadOption.ReadLargeOnDemand);
+            DicomPixelData dicomPixelData = dicomFile.GetPixelDataAndValidateFrames(new[] { request.FrameNumber });
+
+            Stream resultStream = await ConvertToImage(dicomFile, request.FrameNumber, returnHeader.MediaType.ToString(), cancellationToken);
+            string outputContentType = returnHeader.MediaType.ToString();
+
+            sw.Stop();
+            _logger.LogInformation("Render from dicom to {OutputContentType}, uncompressed file size was {UncompressedFrameSize}, output frame size is {OutputFrameSize} and took {ElapsedMilliseconds} ms", outputContentType, stream.Length, resultStream.Length, sw.ElapsedMilliseconds);
+
+            return new RetrieveRenderedResponse(resultStream, resultStream.Length, outputContentType);
+        }
+
+        catch (DataStoreException e)
+        {
+            // Log request details associated with exception. Note that the details are not for the store call that failed but for the request only.
+            _logger.LogError(e, "Error retrieving dicom resource to render");
+            throw;
+        }
+
+    }
+
+    private async Task<Stream> ConvertToImage(DicomFile dicomFile, int frameNumber, string mediaType, CancellationToken cancellationToken)
+    {
+        try
+        {
+            DicomImage dicomImage = new DicomImage(dicomFile.Dataset);
+            using var img = dicomImage.RenderImage(frameNumber);
+            using var sharpImage = img.AsSharpImage();
+            MemoryStream resultStream = _recyclableMemoryStreamManager.GetStream();
+            await sharpImage.SaveAsJpegAsync(resultStream, new SixLabors.ImageSharp.Formats.Jpeg.JpegEncoder(), cancellationToken: cancellationToken);
+            resultStream.Position = 0;
+
+            return resultStream;
+        }
+        catch (Exception e)
```

Comment:
same here. catch is not necessary

Reply:
So I asked him to add this since we could get any exception from Fo-dicom and Six labos. So I thought we could handle it and throw our exception instead of unhandling it. This is somewhat what we do in Transcode

Reply:
Some of the dicom image rendering could throw specifc exceptions from (we saw some when we were testing as well) so i think it is better to catch these and throw our own exception